src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -5286,6 +5286,12 @@ def _subroutine_handler(self, node, parent):
                 f"PSyclone does not support routines that contain one or more "
                 f"ENTRY statements but found '{entry_stmts[0]}'")
 
+        # If the parent of this subroutine is a FileContainer, then we need
+        # to create its symbol and store it there. No visibility information
+        # is available since we're not contained in module.
+        if isinstance(parent, FileContainer):
+            _process_routine_symbols(node, parent.symbol_table, {})
+
```

Comment:
I hadn't thought about this at all for modules, since my assumption had been that modules in files currently just "work", but I can look into this as well - do you want this as part of this PR or can we leave it until later?